app/src/main/java/org/simple/clinic/registration/facility/RegistrationFacilitySelectionInit.kt

```diff
@@ -1,12 +1,35 @@
 package org.simple.clinic.registration.facility
 
 import com.spotify.mobius.First
-import com.spotify.mobius.First.first
 import com.spotify.mobius.Init
+import org.simple.clinic.mobius.first
+import org.simple.clinic.registration.RegistrationConfig
+import java.time.Duration
 
-class RegistrationFacilitySelectionInit: Init<RegistrationFacilitySelectionModel, RegistrationFacilitySelectionEffect> {
+class RegistrationFacilitySelectionInit(
+    private val locationUpdateInterval: Duration,
```

Comment:
If these parameters are part of `RegistrationConfig`, couldn't we have injected it directly in effect handler? 🤔

Reply:
We could have, except I felt in this case that the effect handler should not be in charge of making the decisions around what the location update interval could be and it should be provided that via the effect.

I think the actual problem here is that we do not inject the delegate which requires us to manually construct all these things in the screen. Something to consider in the back of our mind to see if we _can_ inject delegates.